### ISSUE TITLE
unsupported `BuiltinType` kind 104 for type `__mfp8`

### ISSUE THREAD

**Issue by enum-class** [immunant/c2rust#1252]
On macos with sequoia 15.5 and `Apple clang version 17.0.0 (clang-1700.0.13.5)` for **genann** subject, I run `intercept-build make` and then run 
`./target/debug/c2rust transpile --emit-build-files ~/Documents/genann/compile_commands.json` and face this warning and error:

```
warning: c2rust: Encountered unsupported BuiltinType kind 104 for type __mfp8
1 warning generated.
Transpiling example2.c
thread 'main' panicked at 'Type conversion not implemented for TagTypeUnknown expecting 3', c2rust-transpile/src/c_ast/conversion.rs:833:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I did not face this issue on linux, Could you please help me with this ?

**Comment by rdydygin** [comment]
Same error while transpiling lighttpd 1.4 on c2rust git version. Probably similar to #1209.

```
warning: c2rust: Encountered unsupported BuiltinType kind 104 for type __mfp8
5 warnings generated.
Transpiling lemon.c

thread 'main' panicked at c2rust-transpile/src/c_ast/conversion.rs:833:22:
Type conversion not implemented for TagTypeUnknown expecting 3
stack backtrace:
   0: __rustc::rust_begin_unwind
   1: core::panicking::panic_fmt
   2: c2rust_transpile::c_ast::conversion::ConversionContext::visit_node
   3: c2rust_transpile::c_ast::conversion::ConversionContext::new
   4: c2rust_transpile::transpile_single
   5: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
   6: c2rust_transpile::transpile
   7: c2rust_transpile::main
```

Tried with this, and system llvm (17.0.0), both fail with same error.
```
Homebrew clang version 20.1.7
Target: arm64-apple-darwin24.5.0
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/llvm/20.1.7/bin
Configuration file: /opt/homebrew/etc/clang/arm64-apple-darwin24.cfg
```

**Comment by jacob-greenfield-ll** [comment]
I can reproduce this as well:

```
$ echo 'int main() { return 0; }' >foo.c
$ RUST_BACKTRACE=full c2rust transpile foo.c
warning: c2rust: Encountered unsupported BuiltinType kind 104 for type __mfp8
1 warning generated.
Transpiling foo.c

thread 'main' panicked at c2rust-transpile/src/c_ast/conversion.rs:849:22:
Type conversion not implemented for TagTypeUnknown expecting 3
stack backtrace:
   0:        0x104a0d2b8 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h2c3f7b4c4ce00f07
   1:        0x104a28478 - core::fmt::write::h1dbafa36e52e01c5
   2:        0x104a0a3cc - std::io::Write::write_fmt::he72d49f3084ee601
   3:        0x104a0d16c - std::sys::backtrace::BacktraceLock::print::h1fb87370474572ed
   4:        0x104a0e4a8 - std::panicking::default_hook::{{closure}}::h195a9b2c829547eb
   5:        0x104a0e2f8 - std::panicking::default_hook::h18c3aa3e3a3584d5
   6:        0x104a0ef88 - std::panicking::rust_panic_with_hook::h02a9fa3cad928562
   7:        0x104a0ebb4 - std::panicking::begin_panic_handler::{{closure}}::hd1cc56578f819958
   8:        0x104a0d77c - std::sys::backtrace::__rust_end_short_backtrace::h52c1e479035e4bc4
   9:        0x104a0e85c - __rustc[4794b31dd7191200]::rust_begin_unwind
  10:        0x104a4dc78 - core::panicking::panic_fmt::heec96bfc27e6c546
  11:        0x104613674 - c2rust_transpile::c_ast::conversion::ConversionContext::visit_node::h9ea0e1de31378e7d
  12:        0x10460d6e0 - c2rust_transpile::c_ast::conversion::ConversionContext::new::h0d66a3761819ec7d
  13:        0x10463d00c - c2rust_transpile::transpile_single::h46223189964a4efd
  14:        0x1045aca40 - <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter::h8043604a70eb57d9
  15:        0x10463a718 - c2rust_transpile::transpile::hec9076a044eda856
  16:        0x10453454c - c2rust_transpile::main::h8fed0123c922eef2
  17:        0x10453b1c8 - std::sys::backtrace::__rust_begin_short_backtrace::he92685d955e32d8a
  18:        0x10453c79c - std::rt::lang_start::{{closure}}::hfb3d77469a72c7d6
  19:        0x104a05900 - std::rt::lang_start_internal::ha7d36e169a6a9a91
  20:        0x104538550 - _main
```

Built via `LLVM_CONFIG_PATH=/opt/homebrew/opt/llvm/bin/llvm-config cargo +stable install --path c2rust`

macOS 15.5 (arm64)
Homebrew clang 20.1.8
c2rust commit be24dbd2fe090f6fa1574db4b22b55a21a501aea
Stable `rustc 1.88.0 (6b00bc388 2025-06-23)`
